src/python/gem5/resources/workload.py

```diff
@@ -24,219 +24,55 @@
+    class is merely a wrapper for it.
+    """
+    warn(
+        "The `CustomWorkload` class is deprecated. Please use "
+        "`WorkloadResource` instead."
     )
+    return WorkloadResource(function=function, parameters=parameters)
 
-    board.set_workload(workload)
-    ```
-
-    The above is the equivalent of:
 
-    ```py
-    board.set_se_binary_workload(
-        binary = obtain_resource("x86-print-this"),
-        arguments = ["hello", 6],
-    )
-    ```
-
-    Notes
-    -----
-    This class should not be used directly. Please use `Workload` or
-    `CustomWorkload`.
+def Workload(
+    workload_name: str,
+    resource_directory: Optional[str] = None,
+    resource_version: Optional[str] = None,
+    clients: Optional[List] = None,
+    gem5_version: Optional[str] = core.gem5Version,
+):
     """
+    This function was created to maintain backwards compability for v23.0.0
+    and prior releases of gem5 where `Workload` was a class.
 
-    def __init__(self, function: str, parameters: Dict[str, Any]) -> None:
-        self._func = function
-        self._params = parameters
-
-    def get_function_str(self) -> str:
-        """
-        Returns the name of the workload function to be run.
-
-        This function is called via the AbstractBoard's `set_workload`
-        function. The parameters from the `get_parameters` function are passed
-        to this function.
-        """
-        return self._func
-
-    def get_parameters(self) -> Dict[str, Any]:
-        """
-        Returns a dictionary mapping the workload parameters to their values.
-
-        These parameters are passed to the function specified by
-        `get_function_str` via the AbstractBoard's `set_workload` function.
-        """
-        return self._params
-
-    def set_parameter(self, parameter: str, value: Any) -> None:
-        """
-        Used to set or override a workload parameter
-
-        :param parameter: The parameter of the function to set.
-        :param value: The value to set to the parameter.
-        """
-        self._params[parameter] = value
-
-
-class CustomWorkload(AbstractWorkload):
+    In the interests of gem5-resource specialization, the `Workload` class
+    has been dropped. Instead users are advized to use the `obtain_resource`
+    function which will return the correct `AbstractResource` implementation.
+    This function (disguised as a class) wraps this function.
     """
-    A workload specified locally (i.e., not via gem5-resources as with the
-    `Workload` class). Here the user specifies the function and the parameters
-    to be passed.
-
-    Usage
-    -----
-
-    ```py
-    workload = CustomWorkload(
-        function = "set_se_binary_workload",
-        parameters = {
-            "binary" : obtain_resource("x86-print-this"),
-            "arguments" : ["hello", 6]
-        },
+    warn(
+        "`Workload` has been deprecated. Please use the `obtain_resource` "
+        "function instead."
```

Comment:
Let's do people a favor here and construct the warn to say explicitly what call they should be making instead. How about:

```suggestion
        "`Workload` has been deprecated. Please use the `obtain_resource` "
        "function instead:\n\n"
        "```\n"
        "from gem5.resources.resource import obtain_resource\n"
        "workload = obtain_resource(\n"
        f"    resource_id=\"{workload_name}\",\n"
        f"    resource_directory=\"{resource_directory}\",\n"
        f"    gem5_version=\"{gem5_version}\",\n"
        f"    clients={clients},\n"
        f"    resource_version={resource_version},\n"
        ")\n"
        "```"
```